src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -2571,22 +2580,34 @@ _mongoc_delete_one_or_many (mongoc_collection_t *collection,
    BSON_ASSERT_PARAM (selector);
    BSON_ASSERT (bson_empty0 (reply));
 
+   /* TODO: This function has historically used `extra` for top-level, command
+    * options. That is inconsistent with the update function, which uses `extra`
+    * for statement-level options. We will keep the original behavior for BC
+    * reasons, but this should ultimately be addressed by CDRIVER-4306. */
```

Comment:
See: https://jira.mongodb.org/browse/CDRIVER-4306

I decided to create `cmd_opts` since it's more descriptive, allows `extra` to remain a `const bson_t *`, and will play nicer with any future refactoring to this method.